ping/ping6_common.c

```diff
@@ -832,6 +932,71 @@ int ping6_parse_reply(struct ping_rts *rts, socket_st *sock,
 			fflush(stdout);
 			return 0;
 		}
+	} else if (icmph->icmp6_type == ICMPV6_EXT_ECHO_REPLY) {
+		if (!rts->multicast &&
+		    memcmp(&from->sin6_addr.s6_addr, &rts->whereto6.sin6_addr.s6_addr, 16))
+			return 1;
+		if (!is_ours(rts, sock, icmph->icmp6_id))
+			return 1;
+
+		sequence = ntohs(icmph->icmp6_seq);
+		state = icmph->icmp6_seq & 0xe0;
+		printf("Interface: %s\n", rts->interface);
+		switch (icmph->icmp6_code) {
+			case 1:
+				printf("Error: Malformed Query\n");
+				break;
+			case 2:
+				printf("Error: No Such Interface\n");
+				break;
+			case 3:
+				printf("Error: No Such Table Entry\n");
+				break;
+			case 4:
+				printf("Error: Multiple Interfaces Satisfy Query\n");
+				break;
+			default:
```

Comment:
An invalid code (5 or above) will result in no error, and no output from the "code == 0" case below.  This will likely be fairly confusing.  Sure, it shouldn't happen, but, let's try to confuse the user less if it does occur.